bin/mge_integrator.py

```diff
@@ -899,150 +43,137 @@ def main():
-    (contig_ismge, contig_inmge) = is_grouping(mge_data)
+    # Parsing PaliDIS results and removing redundancy with ISEScan
+    if os.path.exists(args.pal_tsv):
+        (mge_data, itr_sites) = palidis_process.palids_parser(
+            args.pal_tsv, inv_names_equiv, mge_data, itr_sites
+        )
 
-    # Removing overlapping ISs
-    (mge_data, mge_nuc) = is_overlap(
-        contig_ismge, 
-        mge_data, 
-        mge_nuc,
-    )
+    # Parsing geNomad results
+    (mge_data) = genomad_parser.genomad_viral(args.geno_out, mge_data)
+    (mge_data) = genomad_parser.plasmids_parser(args.geno_plas, mge_data)
 
-    # Reporting overlapping integrons
-    int_overlap(contig_inmge, mge_data)
+    # Parsing VIRIfy results and solving redundancy with geNomad
+    if os.path.exists(args.virify_out):
+        (mge_data, virify_prots) = virify_process.virify_reader(
+            args.virify_out, inv_names_equiv, mge_data
+        )
+    else:
+        virify_prots = {}
 
-    # Saving protein coordinates
-    (contig_prots, prots_coord) = location_parser(cds_loc)
+    ## Overlapping report for long MGEs
+    # Collecting integrons, virus and plasmids predicted per contig
+    overlap_finder.overlap_report(mge_data, names_equiv)
 
-    # Saving mobilome proteins
-    (
-        mge_proteins, 
-        mob_proteome, 
-        contigs_elements, 
-        proteins_mge,
-    ) = mob_cds(
-        mge_data, 
-        names_equiv, 
-        contig_prots,
-        user_gff,
-        prots_coord,
+    ## Catching proteins on the mobilome and MGEs QC
+    (mge_data, contigs_elements, proteins_mge) = cds_locator.location_parser(
+        args.pkka_gff, mge_data
     )
 
+    ## Storing extra annotation results
     # Parsing mobileOG results
-    mog_annot = mobileog_parser(mog_results, mob_proteome)
-
-    # Filtering out low-quality results
-    (mge_data , contigs_elements ) = quality_filter(mge_data, mge_proteins, contigs_elements)
-
-    # Generating the final output
-    writing_gff(
-        cds_loc, 
-        names_equiv, 
-        inv_names_equiv, 
-        contigs_elements, 
-        itr_sites, 
+    mog_annot = mobileog_process.mobileog_parser(args.mog_tsv)
+
+    # Parsing CRISPRCasFinder results
+    if os.stat(args.crispr_out).st_size > 0:
+        crispr_annot = crispr_process.crispr_parser(args.crispr_out, args.pkka_gff)
+    else:
+        crispr_annot = {}
+
+    # Adding the mobilome annotation to the GFF file
+    integrator_process.gff_writer(
+        args.pkka_gff,
+        names_equiv,
+        contigs_elements,
+        itr_sites,
         icf_dr,
         attC_site,
-        mge_data, 
-        mge_nuc, 
-        proteins_mge, 
+        mge_data,
+        proteins_mge,
         mog_annot,
-        user_gff,
+        crispr_annot,
+        virify_prots,
     )
 
 
 if __name__ == "__main__":
-    main()
-
+    import argparse
+    import os.path
+    import sys
+
+    sys.path.append("./python_modules")
```

Comment:
This is not portable.

Reply:
But this is the directories structure of the repo. What would be the right way to refer the location of the python_modules?